core/src/components/segment/segment.tsx

```diff
@@ -51,17 +54,29 @@ export class Segment implements ComponentInterface {
   @Prop({ mutable: true }) value?: string | null;
 
   @Watch('value')
-  protected valueChanged(value: string | undefined) {
-    if (this.didInit) {
-      this.ionChange.emit({ value });
+  protected valueChanged(value: string | undefined, oldValue: string | undefined | null) {
+    this.ionSelect.emit({ value });
+    if (oldValue !== '' || this.didInit) {
```

Comment:
Why is needed to check for oldValue?

Reply:
When binding values in Angular using `ngModel`, `ion-segment` value is initialized to `null` and then to the `ngModel` value. We had an old fix in place, but it was causing issues in https://github.com/ionic-team/ionic/issues/20500.